llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -7095,8 +7095,7 @@ SDValue DAGCombiner::visitAND(SDNode *N) {
     // fold (and (masked_load) (splat_vec (x, ...))) to zext_masked_load
     auto *MLoad = dyn_cast<MaskedLoadSDNode>(N0);
     ConstantSDNode *Splat = isConstOrConstSplat(N1, true, true);
-    if (MLoad && MLoad->getExtensionType() == ISD::EXTLOAD && Splat &&
-        N1.hasOneUse()) {
```

Comment:
Should it be checking N0.hasOneUse()? (i.e the load)

Reply:
If I add that check it causes regression in CodeGen/AArch64/sve-load-compare-store.ll and CodeGen/Thumb2/mve-masked-load.ll

Reply:
An extra instruction in multi_user_zext (mve-masked-load.ll) and two extra in sve_load_compare_store (sve-load-compare-store.ll)

Reply:
You can see from the code the transformation has explicit handling for this.  The original load is an any-extend and so the transformation is simply promoting this to a zero-extend which can be used by all uses.